lib/ClangImporter/ImportDecl.cpp

```diff
@@ -5756,7 +5756,8 @@ SwiftDeclConverter::getImplicitProperty(ImportedName importedName,
         assert(clangEnum.value()->getIntegerType()->getCanonicalTypeInternal() ==
                typedefType->getCanonicalTypeInternal());
         if (auto swiftEnum = Impl.importDecl(*clangEnum, Impl.CurrentVersion)) {
-          importedType = {cast<NominalTypeDecl>(swiftEnum)->getDeclaredType(), false};
```

Comment:
Under the right conditions, there are `TypeAliasDecls`, not `NominalTypeDecl`s.

Reply:
Looked around a bit, if this were Objective-C, and we didn't need to play this find the anonymous tag decl for this typedef game, we are calling `getDeclaredInterfaceType`: https://github.com/apple/swift/blob/main/lib/ClangImporter/ImportType.cpp#L997. Unless something breaks, that is convincing enough for me.

Reply:
Do we have any evidence to suggest that we should do `cast<NominalTypeDecl>(swiftEnum)->getDeclaredType()` for C/ObjC interop and `cast<TypeDecl>(swiftEnum)->getDeclaredInterfaceType()` for C++-Interop? I think it'd be best to keep things uniform between th two but Just wondering what would work best for either case.

Reply:
I don't think there is any reason for them to be different. When I said "if this were Objective-C", I didn't really mean as a consequence of the Swift compiler behaving differently with cxx-interop enabled. Since NS_OPTION is defined differently in C++-mode, we have to go from the backing typedef, to the anonymous enum, import that, then get its type. If we had the enum in the first place (as we would if NS_OPTION in C++ wasn't so weird), we'd return `getDeclaredInterfaceType` at the line I linked, so I think that is the appropriate thing to do.

As an aside, the different between the two methods seems to be related to generics, and to my knowledge, C/Objective-C/C++ enums can't be generic, so I can't imagine why they'd be imported as such. I don't think `getDeclaredType` was chosen for any particular reason, but I think @zoecarver wrote it, so I'll let him chime in if he sees fit 🙂